# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../fam/fam-2.7.0-gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Fix to build with gcc >= 4.3

diff -ruN fam-2.7.0-dnotify-gcc41/include/BTree.h fam-2.7.0-dnotify-gcc43/include/BTree.h
--- fam-2.7.0-dnotify-gcc41/include/BTree.h	2003-01-20 05:22:30.000000000 +0100
+++ fam-2.7.0-dnotify-gcc43/include/BTree.h	2010-08-29 14:06:22.000000000 +0200
@@ -24,6 +24,7 @@
 #define BTree_included
 
 #include "Boolean.h"
+#include <cstdlib>
 
 //  This is an in-core B-Tree implementation.
 //
diff -ruN fam-2.7.0-dnotify-gcc41/lib/Client.c++ fam-2.7.0-dnotify-gcc43/lib/Client.c++
--- fam-2.7.0-dnotify-gcc41/lib/Client.c++	2003-01-18 15:18:12.000000000 +0100
+++ fam-2.7.0-dnotify-gcc43/lib/Client.c++	2010-08-29 14:06:22.000000000 +0200
@@ -34,7 +34,7 @@
 #include <syslog.h>
 #include <errno.h>
 
-#include <iostream.h>
+#include <iostream>
 
 #include "fam.h"
 #include "Client.h"
diff -ruN fam-2.7.0-dnotify-gcc41/src/DNotify.c++ fam-2.7.0-dnotify-gcc43/src/DNotify.c++
--- fam-2.7.0-dnotify-gcc41/src/DNotify.c++	2010-08-29 13:09:03.000000000 +0200
+++ fam-2.7.0-dnotify-gcc43/src/DNotify.c++	2010-08-29 14:06:22.000000000 +0200
@@ -31,6 +31,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <libgen.h>
+#include <cstdlib>
 
 #include "DNotify.h"
 
diff -ruN fam-2.7.0-dnotify-gcc41/src/IMon.c++ fam-2.7.0-dnotify-gcc43/src/IMon.c++
--- fam-2.7.0-dnotify-gcc41/src/IMon.c++	2003-01-18 15:18:12.000000000 +0100
+++ fam-2.7.0-dnotify-gcc43/src/IMon.c++	2010-08-29 14:06:22.000000000 +0200
@@ -40,7 +40,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <memory>
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
diff -ruN fam-2.7.0-dnotify-gcc41/src/Interest.h fam-2.7.0-dnotify-gcc43/src/Interest.h
--- fam-2.7.0-dnotify-gcc41/src/Interest.h	2010-08-29 13:09:03.000000000 +0200
+++ fam-2.7.0-dnotify-gcc43/src/Interest.h	2010-08-29 14:06:22.000000000 +0200
@@ -29,6 +29,7 @@
 #include <netinet/in.h>  //  for in_addr
 
 #include "Boolean.h"
+#include <cstdlib>
 
 class Event;
 class FileSystem;
